src/sinks/datadog/metrics.rs

```diff
@@ -187,13 +187,16 @@ impl SinkConfig for DatadogConfig {
             HttpBatchService::new(client, move |request| ready(sink.build_request(request))),
         );
 
-        let buffer = PartitionBuffer::new(MetricsBuffer::<DatadogMetricNormalize>::new(batch.size));
+        let buffer = PartitionBuffer::new(MetricsBuffer::new(batch.size));
+        let mut normalizer = MetricNormalizer::<DatadogMetricNormalize>::default();
 
         let svc_sink = PartitionBatchSink::new(svc, buffer, batch.timeout, cx.acker())
             .sink_map_err(|error| error!(message = "Fatal datadog metric sink error.", %error))
             .with_flat_map(move |event: Event| {
-                let ep = DatadogEndpoint::from_metric(&event);
-                stream::iter(Some(PartitionInnerBuffer::new(event, ep))).map(Ok)
+                stream::iter(normalizer.apply(event).map(|event| {
+                    let ep = DatadogEndpoint::from_metric(&event);
+                    Ok(PartitionInnerBuffer::new(event, ep))
```

Comment:
```suggestion
                    let endpoint = DatadogEndpoint::from_metric(&event);
                    Ok(PartitionInnerBuffer::new(event, endpoint))
```

Reply:
I copied that naming over from the existing code, but you're right, the full name is better since I was rewriting the code anyways. Fixed.